ipa-core/src/test_fixture/ipa.rs

```diff
@@ -245,75 +244,25 @@ pub async fn test_oprf_ipa<F>(
     semi_honest::AdditiveShare<F>: Serializable,
+    Replicated<F>: Serializable,
 {
     use crate::{
-        ff::{
-            boolean::Boolean,
-            boolean_array::{BA20, BA3, BA5, BA8},
-            Field,
-        },
-        protocol::{
-            basics::ShareKnownValue,
-            ipa_prf::prf_sharding::{
-                attribution_and_capping_and_aggregation, compute_histogram_of_users_with_row_count,
-                PrfShardedIpaInputRow,
-            },
-        },
-        report::EventType,
-        secret_sharing::SharedValue,
+        ff::boolean_array::{BA20, BA3, BA5, BA8},
+        protocol::ipa_prf::oprf_ipa,
+        report::OprfReport,
         test_fixture::Runner,
     };
 
-    let user_cap: i32 = config.per_user_credit_cap.try_into().unwrap();
-    assert!(
-        user_cap & (user_cap - 1) == 0,
-        "This code only works for a user cap which is a power of 2"
-    );
-
     //TODO(richaj) This manual sorting will be removed once we have the PRF sharding in place
     records.sort_by(|a, b| b.user_id.cmp(&a.user_id));
 
-    let histogram = compute_histogram_of_users_with_row_count(&records);
-    let ref_to_histogram = &histogram;
-
-    let result: Vec<F> = world
+    let result: Vec<_> = world
         .semi_honest(
             records.into_iter(),
-            |ctx, input_rows: Vec<OprfReport<BA20, BA8, BA3>>| async move {
-                let sharded_input = input_rows
-                    .into_iter()
-                    .map(|single_row| {
-                        let is_trigger_bit_share = if single_row.event_type == EventType::Trigger {
-                            Replicated::share_known_value(&ctx, Boolean::ONE)
-                        } else {
-                            Replicated::share_known_value(&ctx, Boolean::ZERO)
-                        };
-                        PrfShardedIpaInputRow {
-                            prf_of_match_key: single_row.mk_oprf,
-                            is_trigger_bit: is_trigger_bit_share,
-                            breakdown_key: single_row.breakdown_key,
-                            trigger_value: single_row.trigger_value,
-                            timestamp: single_row.timestamp,
-                        }
-                    })
-                    .collect::<Vec<_>>();
-
-                attribution_and_capping_and_aggregation::<
-                    _,
-                    BA8,  // BreakdownKey,
-                    BA3,  // TriggerValue,
-                    BA20, // Timestamp,
-                    BA5,  // Saturating Sum
-                    Replicated<F>,
-                    F,
-                >(
-                    ctx,
-                    sharded_input,
-                    config.attribution_window_seconds,
-                    ref_to_histogram,
-                )
-                .await
-                .unwrap()
+            |ctx, input_rows: Vec<OprfReport<_, _, _>>| async move {
```

Comment:
I am unable to get it to compile when I run cargo bench for this. It complains of intoiter not being implemented but it is actually implemented.

Reply:
you likely need `text-fixture` feature because the implementation is inside that module